src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/linalg/FourierLinearOperator.java

```diff
@@ -0,0 +1,99 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.linalg;
+
+import org.apache.commons.math3.exception.DimensionMismatchException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.jtransforms.fft.DoubleFFT_1D;
+
+import javax.annotation.Nonnull;
+import java.util.stream.IntStream;
+
+import static com.google.common.math.IntMath.isPowerOfTwo;
+import static org.broadinstitute.hellbender.utils.GATKProtectedMathUtils.nearestNeighborUniform1DInterpolate;
+import static org.broadinstitute.hellbender.utils.GATKProtectedMathUtils.smallestPowerOfTwoGreaterThan;
+
+/**
+ * A circulant linear operator specified by its eigenvalues in the Fourier space.
```

Comment:
I think that this javadoc could be more useful if it provided some brief mathematical exposition on circulant matrices and the DFT matrix.  At least enough so that the definition of the fourierFactors is clear.